bedms/train.py

```diff
@@ -0,0 +1,242 @@
+from sklearn.metrics import (
+    precision_score,
+    recall_score,
+    f1_score,
+)
+import yaml
+from .utils_train import (
+    load_from_dir,
+    accumulate_data,
+    training_encoding,
+    data_loader,
+    train_model,
+    plot_learning_curve,
+    model_testing,
+    plot_confusion_matrix,
+    auc_roc_curve,
+)
+from .const import PROJECT_NAME
+from .model import BoWSTModel
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(PROJECT_NAME)
+
+
+class AttrStandardizerTrainer:
+    """
+    This is the training class responsible for
+    managing the training process for the standardizer model.
+    """
+
+    def __init__(self, config: str) -> None:
+        """
+        Initializes the TrainStandardizer object with the given configuration.
+
+        :param str config: Path to the config file which has the training parameters provided by the user.
+        """
+        self.label_encoder = None
+        self.vectorizer = None
+        self.train_loader = None
+        self.val_loader = None
+        self.test_loader = None
+        self.output_size = None
+        self.criterion = None
+        self.train_accuracies = None
+        self.val_accuracies = None
+        self.train_losses = None
+        self.val_losses = None
+        self.model = None
+        self.fpr = None
+        self.tpr = None
+        self.roc_auc = None
+        self.all_labels = None
+        self.all_preds = None
+
+        with open(config, "r") as file:
+            self.config = yaml.safe_load(file)
+
+    def load_encode_data(self) -> None:
+        """
+        Loads and prepares the encoded training, testing and validation datasets.
+        """
+        values_files_list = load_from_dir(self.config["dataset"]["values_dir_pth"])
+        headers_files_list = load_from_dir(self.config["dataset"]["headers_dir_pth"])
+
+        if len(values_files_list) != len(headers_files_list):
+            logger.error(
+                f"Mismatch in number of value files ({len(values_files_list)}) \
+                and header files ({len(headers_files_list)})"
+            )
+            return
+
+        total_files = len(values_files_list)
+
+        paired_files = list(zip(values_files_list, headers_files_list))
+
+        train_size = self.config["data_split"]["train_set"]
+        test_size = self.config["data_split"]["test_set"]
+        val_size = self.config["data_split"]["val_set"]
+
+        if train_size + val_size + test_size > total_files:
+            logger.error(
+                f"Data split sizes exceed total number of files: "
+                f"train({train_size}) + val({val_size}) + \
+                test({test_size}) > total_files({total_files})"
+            )
+            return
+
+        train_files = paired_files[:train_size]
+        val_files = paired_files[train_size : train_size + val_size]
+        test_files = paired_files[
+            train_size + val_size : train_size + val_size + test_size
+        ]
```

Comment:
Then down here you can say:
```python
num_train_files = int(train_size * len(paired_files))
train_files = paired_files[:num_train_files]
```

Also now looking at this I'd make sure to shuffle `paired_files` before calling this code; that way you aren't getting big class imbalances in your training and test/validation sets